packages/extension-browser/src/content-script/connector.ts

```diff
@@ -172,7 +176,69 @@ export default class WebExtensionConnector implements IConnector {
         });
     }
 
+    private needsToRunInPage(source: string) {
+        return source.includes('/*RunInPageContext*/');
+    }
+
+    private scriptsCounter: number = 0;
+
+    /**
+     * Runs a script in the website context.
+     *
+     * By default, `eval` runs the scripts in a different context
+     * but, some scripts, needs to run in the same context
+     * of the website.
+     */
+    private evaluateInPage(source: string): Promise<any> {
+        return new Promise((resolve) => {
+            const script = document.createElement('script');
+            const config = {
+                attributes: true,
+                childList: false,
+                subtree: false
+            };
+
+            script.id = `webhint-evaluate-script-element-${this.scriptsCounter++}`;
+
+            const callback = (mutationsList: MutationRecord[], observer: MutationObserver) => {
+                mutationsList.forEach((mutation: MutationRecord) => {
+                    if (mutation.type !== 'attributes' && mutation.attributeName !== 'data-result') {
+                        return;
+                    }
+
+                    const result = JSON.parse(script.getAttribute('data-result')!);
+
+                    document.body.removeChild(script);
+                    observer.disconnect();
+
+                    resolve(result);
+                });
+            };
+
+            const observer = new MutationObserver(callback);
+
+            observer.observe(script, config);
+
+            document.body.appendChild(script);
+            script.textContent = `(async () => {
+                const result = await ${source}
+const scriptElement = document.getElementById('${script.id}');
```

Comment:
Not essential, but if you want to avoid generating a unique ID, you can use [`document.currentScript`](https://developer.mozilla.org/en-US/docs/Web/API/Document/currentScript) (but you’ll probably need to grab a reference before the `await` call).

Reply:
done!